src/decorators/restrict.js

```diff
@@ -0,0 +1,11 @@
+export default function restrict(func) {
+	return (target, key, descriptor) => {
+		const wrapped = descriptor.value
+		descriptor.value = function(...fnargs) {
+			if (func(fnargs[0])) {
```

Comment:
According to issue #3, it should work with a function that returns a promise too.